mirrord-layer/src/lib.rs

```diff
@@ -425,11 +434,62 @@ fn enable_hooks() {
         )
         .unwrap();
 
-    interceptor
-        .replace(
-            Module::find_export_by_name(None, "accept").unwrap(),
-            NativePointer(accept_detour as *mut c_void),
+    if let Some(addr) = Module::find_export_by_name(None, "accept") {
+        match interceptor.replace(
+            addr,
+            NativePointer(accept4_detour as *mut c_void),
             NativePointer(std::ptr::null_mut::<c_void>()),
-        )
-        .unwrap();
+        ) {
+            Err(e) => match e {
+                Error::InterceptorAlreadyReplaced => {
+                    error!("accept already replaced");
+                }
+                _ => {
+                    error!("accept error: {:?}", e);
+                }
+            },
+            Ok(_) => {
+                debug!("accept hooked");
+            }
+        }
```

Comment:
We're not doing proper error handling yet, so you can drop the `match` in favor of `unwrap`.

```rust
interceptor.replace(...).unwrap();
```

If you're doing this to help you debug, `Result` has a method (on nightly) to inspect errors: [inspect_err](https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect_err). Just need to enable the feature `#![feature(result_option_inspect)]`.

Reply:
That's actually my suggestion because on some systems accept4 can be same as accept and uv_accept4 can lead to accept4 directly, so we might encounter a situation where `Error::InterceptorAlreadyReplaced` is an okay situation.